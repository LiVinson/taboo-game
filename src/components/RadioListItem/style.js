import styled from 'styled-components'

export const StyledRadioListItem = styled.li`
	display: flex;
	align-items: center;
	width: 100%;
	&:not(:last-child) {
		border-bottom: 1px solid ${(props) => props.theme.color.lightGray2};
	}

	/*selects last list item, and removes the bottom padding from label since it's alrady provided by parent taboo card */
	&:last-child > label {
		padding-bottom: 0;
	}
	/*selects first list item, and removes the top padding from label since it's alrady provided by parent taboo card */

	&:first-child > label {
		padding-top: 0;
	}
`

//hidden so only label displays. Selecting a label results in additional styles being applied
export const RadioInput = styled.input`
	display: none;
	&:checked + label {
		/* font-size: 2.2rem; */
		font-weight: 500;
		/* star that displays when the radio button corresponding to this label is checked */
		&:before {
			content: '>';
			width: 1rem;
			display: inline-block;
			padding-right: 2rem;
			color: ${(props) => props.theme.color.accent2};
		}
	}
`

export const RadioLabel = styled.label`
	display: inline-block;
	font-size: 2rem;
	width: 100%;
	padding: 1rem 0rem;
	color: ${(props) => props.theme.color.grayDark2};
`
