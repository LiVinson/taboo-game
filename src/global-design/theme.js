const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
