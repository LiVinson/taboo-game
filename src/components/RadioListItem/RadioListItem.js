import React from 'react'
import PropTypes from 'prop-types'
import { StyledRadioListItem, RadioInput, RadioLabel } from './style'

//Takes in taboo word, it's index, status, and a cb function that is called when user selects a different word in the list
//Returns a list item with a hidden radio button and label.
const RadioListItem = ({ word, index, status, checked, onChange }) => {
	return (
		<StyledRadioListItem>
			<RadioInput
				checked={checked}
				type="radio"
				name={status}
				value={index}
				id={word}
				onChange={(e) => {
					onChange(e.target.value, status)
				}}
			/>
			<RadioLabel htmlFor={word}>{word}</RadioLabel>
		</StyledRadioListItem>
	)
}

RadioListItem.propTypes = {
	word: PropTypes.string.isRequired,
	index: PropTypes.number.isRequired,
	status: PropTypes.string.isRequired,
	onChange: PropTypes.func.isRequired,
}
export default RadioListItem
