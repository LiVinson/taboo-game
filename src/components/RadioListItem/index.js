import RadioListItem from "./RadioListItem"

export default RadioListItem