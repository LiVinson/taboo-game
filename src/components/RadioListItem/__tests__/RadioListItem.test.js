import React from 'react'
import renderer from 'react-test-renderer'
import {  mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import RadioListItem from '../RadioListItem'

describe('RadioListItem', () => {
	const defaultProps = {
		word: 'apple',
		index: 0,
		status: 'correct',
		checked: false, 
		onChange: jest.fn(),
	}

	test('Renders correctly', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<RadioListItem {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Selecting radio button fires props.onChange', () => {
		const props = {
			...defaultProps,
		}
		const component = mount(
			<ThemeProvider theme={theme}>
				<RadioListItem {...props} />
			</ThemeProvider>
		)
		// const component = wrapper.dive().find(RadioListItem)
		const radio = component.find('radio')
		const label = component.find('label')
		label.simulate('click', { target: { value: props.index } })

        expect(radio.value).toBe(props.value)
	})
})
