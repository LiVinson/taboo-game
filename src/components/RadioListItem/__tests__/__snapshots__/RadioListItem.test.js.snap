// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioListItem Renders correctly 1`] = `
<li
  className="sc-AxjAm kecFbU"
>
  <input
    checked={false}
    className="sc-AxirZ hHkOEQ"
    id="apple"
    name="correct"
    onChange={[Function]}
    type="radio"
    value={0}
  />
  <label
    className="sc-AxiKw kMXxWh"
    htmlFor="apple"
  >
    apple
  </label>
</li>
`;
